### ISSUE TITLE
SD Automatic1111/Local: Training failed 

### ISSUE THREAD

**Issue by KampfschaFF** [modelscope/facechain#410]
Getting error when i try to train

Log:

--------uuid:  qw
----------work_dir:  E:\stable-diffusion-webui\extensions\facechain\worker_data\qw\ly261666/cv_portrait_model\person2
2023-11-07 14:39:38,531 - modelscope - INFO - Use user-specified model revision: v1.0.0
Weights loaded in 48.1s (send model to cpu: 0.4s, calculate hash: 26.1s, load weights from disk: 0.5s, apply weights to model: 0.4s, move model to device: 20.7s).
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
2023-11-07 14:40:02,317 - modelscope - INFO - PyTorch version 2.0.1+cu118 Found.
2023-11-07 14:40:02,319 - modelscope - INFO - Loading ast index from C:\Users\kff90\.cache\modelscope\ast_indexer
2023-11-07 14:40:02,546 - modelscope - INFO - Loading done! Current index file version is 1.9.4, with md5 c1336cf14d82e9e22e17a24b9f4fca0b and a total number of 945 components indexed
2023-11-07 14:40:06,075 - modelscope - INFO - Use user-specified model revision: v4.0
2023-11-07 14:40:11,770 - modelscope - INFO - Use user-specified model revision: v1.0.1
2023-11-07 14:40:12,310 - modelscope - WARNING - ('PIPELINES', 'skin-retouching-torch', 'skin-retouching-torch') not found in ast index file
2023-11-07 14:40:12,310 - modelscope - INFO - initiate model from C:\Users\kff90\.cache\modelscope\hub\damo\cv_unet_skin_retouching_torch
2023-11-07 14:40:12,310 - modelscope - INFO - initiate model from location C:\Users\kff90\.cache\modelscope\hub\damo\cv_unet_skin_retouching_torch.
2023-11-07 14:40:12,316 - modelscope - WARNING - No preprocessor field found in cfg.
2023-11-07 14:40:12,316 - modelscope - WARNING - No val key and type key found in preprocessor domain of configuration.json file.
2023-11-07 14:40:12,316 - modelscope - WARNING - Cannot find available config to build preprocessor at mode inference, current config: {'model_dir': 'C:\\Users\\kff90\\.cache\\modelscope\\hub\\damo\\cv_unet_skin_retouching_torch'}. trying to build by task and model information.
2023-11-07 14:40:12,316 - modelscope - WARNING - Find task: skin-retouching-torch, model type: None. Insufficient information to build preprocessor, skip building preprocessor
2023-11-07 14:40:16,149 - modelscope - WARNING - Model revision not specified, use revision: v2.0.2
2023-11-07 14:40:18,529 - modelscope - INFO - initiate model from C:\Users\kff90\.cache\modelscope\hub\damo\cv_resnet50_face-detection_retinaface
2023-11-07 14:40:18,529 - modelscope - INFO - initiate model from location C:\Users\kff90\.cache\modelscope\hub\damo\cv_resnet50_face-detection_retinaface.
2023-11-07 14:40:18,534 - modelscope - WARNING - No preprocessor field found in cfg.
2023-11-07 14:40:18,535 - modelscope - WARNING - No val key and type key found in preprocessor domain of configuration.json file.
2023-11-07 14:40:18,535 - modelscope - WARNING - Cannot find available config to build preprocessor at mode inference, current config: {'model_dir': 'C:\\Users\\kff90\\.cache\\modelscope\\hub\\damo\\cv_resnet50_face-detection_retinaface'}. trying to build by task and model information.
2023-11-07 14:40:18,535 - modelscope - WARNING - Find task: face-detection, model type: None. Insufficient information to build preprocessor, skip building preprocessor
2023-11-07 14:40:18,536 - modelscope - INFO - loading model from C:\Users\kff90\.cache\modelscope\hub\damo\cv_resnet50_face-detection_retinaface\pytorch_model.pt
2023-11-07 14:40:18,865 - modelscope - INFO - load model done
2023-11-07 14:40:20.1874127 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2649'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2077256 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2644'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2110708 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2647'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2143447 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2658'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2174462 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2648'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2205092 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2657'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2237298 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2653'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2270753 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2652'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2300452 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2645'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2332813 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2643'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2362374 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2641'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2394560 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2633'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2424080 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2632'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2453965 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2624'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2487899 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2614'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2518094 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2613'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2548426 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2606'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2577850 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2598'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2610677 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2596'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:20.2644035 [W:onnxruntime:, graph.cc:3553 onnxruntime::Graph::CleanUnusedInitializersAndNodeArgs] Removing initializer 'const_fold_opt__2594'. It is not used by any node and should be removed from the model.
2023-11-07 14:40:26,338 - modelscope - INFO - Use user-specified model revision: v1.1
2023-11-07 14:40:26,872 - modelscope - INFO - initiate model from C:\Users\kff90\.cache\modelscope\hub\damo\cv_ddsar_face-detection_iclr23-damofd
2023-11-07 14:40:26,873 - modelscope - INFO - initiate model from location C:\Users\kff90\.cache\modelscope\hub\damo\cv_ddsar_face-detection_iclr23-damofd.
2023-11-07 14:40:26,874 - modelscope - INFO - initialize model from C:\Users\kff90\.cache\modelscope\hub\damo\cv_ddsar_face-detection_iclr23-damofd
Process Process-3:
Traceback (most recent call last):
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\utils\registry.py", line 210, in build_from_cfg
    return obj_cls._instantiate(**args)
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\models\base\base_model.py", line 67, in _instantiate
    return cls(**kwargs)
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\models\cv\face_detection\scrfd\damofd_detect.py", line 31, in __init__
    super().__init__(model_dir, **kwargs)
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\models\cv\face_detection\scrfd\scrfd_detect.py", line 36, in __init__
    from mmdet.models import build_detector
  File "E:\stable-diffusion-webui\venv\lib\site-packages\mmdet\models\__init__.py", line 2, in <module>
    from .backbones import *  # noqa: F401,F403
  File "E:\stable-diffusion-webui\venv\lib\site-packages\mmdet\models\backbones\__init__.py", line 2, in <module>
    from .csp_darknet import CSPDarknet
  File "E:\stable-diffusion-webui\venv\lib\site-packages\mmdet\models\backbones\csp_darknet.py", line 11, in <module>
    from ..utils import CSPLayer
  File "E:\stable-diffusion-webui\venv\lib\site-packages\mmdet\models\utils\__init__.py", line 13, in <module>
    from .point_sample import (get_uncertain_point_coords_with_randomness,
  File "E:\stable-diffusion-webui\venv\lib\site-packages\mmdet\models\utils\point_sample.py", line 3, in <module>
    from mmcv.ops import point_sample
  File "E:\stable-diffusion-webui\venv\lib\site-packages\mmcv\ops\__init__.py", line 2, in <module>
    from .active_rotated_filter import active_rotated_filter
  File "E:\stable-diffusion-webui\venv\lib\site-packages\mmcv\ops\active_rotated_filter.py", line 10, in <module>
    ext_module = ext_loader.load_ext(
  File "E:\stable-diffusion-webui\venv\lib\site-packages\mmcv\utils\ext_loader.py", line 13, in load_ext
    ext = importlib.import_module('mmcv.' + name)
  File "C:\Users\kff90\AppData\Local\Programs\Python\Python310\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ModuleNotFoundError: No module named 'mmcv._ext'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\utils\registry.py", line 212, in build_from_cfg
    return obj_cls(**args)
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\pipelines\cv\face_detection_pipeline.py", line 36, in __init__
    super().__init__(model=model, **kwargs)
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\pipelines\base.py", line 99, in __init__
    self.model = self.initiate_single_model(model)
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\pipelines\base.py", line 53, in initiate_single_model
    return Model.from_pretrained(
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\models\base\base_model.py", line 183, in from_pretrained
    model = build_model(model_cfg, task_name=task_name)
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\models\builder.py", line 35, in build_model
    model = build_from_cfg(
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\utils\registry.py", line 215, in build_from_cfg
    raise type(e)(f'{obj_cls.__name__}: {e}')
ModuleNotFoundError: DamoFdDetect: No module named 'mmcv._ext'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\kff90\AppData\Local\Programs\Python\Python310\lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
  File "C:\Users\kff90\AppData\Local\Programs\Python\Python310\lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "E:\stable-diffusion-webui\extensions\facechain\facechain\inference.py", line 24, in _data_process_fn_process
    Blipv2()(input_img_dir)
  File "E:\stable-diffusion-webui\extensions\facechain\facechain\data_process\preprocessing.py", line 207, in __init__
    self.face_detection = pipeline(task=Tasks.face_detection, model='damo/cv_ddsar_face-detection_iclr23-damofd', model_revision='v1.1')
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\pipelines\builder.py", line 164, in pipeline
    return build_pipeline(cfg, task_name=task)
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\pipelines\builder.py", line 67, in build_pipeline
    return build_from_cfg(
  File "E:\stable-diffusion-webui\venv\lib\site-packages\modelscope\utils\registry.py", line 215, in build_from_cfg
    raise type(e)(f'{obj_cls.__name__}: {e}')
ModuleNotFoundError: FaceDetectionPipeline: DamoFdDetect: No module named 'mmcv._ext'
instance_data_dir E:\stable-diffusion-webui\extensions\facechain\worker_data\qw\training_data\ly261666/cv_portrait_model\person2
The following values were not passed to `accelerate launch` and had defaults used instead:
        `--num_processes` was set to a value of `1`
        `--num_machines` was set to a value of `1`
        `--mixed_precision` was set to a value of `'no'`
        `--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
Traceback (most recent call last):
  File "E:\stable-diffusion-webui\extensions\facechain\facechain\train_text_to_image_lora.py", line 31, in <module>
    import datasets
ModuleNotFoundError: No module named 'datasets'
Traceback (most recent call last):
  File "C:\Users\kff90\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\kff90\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\kff90\AppData\Local\Programs\Python\Python310\Scripts\accelerate.exe\__main__.py", line 7, in <module>
  File "C:\Users\kff90\AppData\Local\Programs\Python\Python310\lib\site-packages\accelerate\commands\accelerate_cli.py", line 47, in main
    args.func(args)
  File "C:\Users\kff90\AppData\Local\Programs\Python\Python310\lib\site-packages\accelerate\commands\launch.py", line 986, in launch_command
    simple_launcher(args)
  File "C:\Users\kff90\AppData\Local\Programs\Python\Python310\lib\site-packages\accelerate\commands\launch.py", line 628, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['C:\\Users\\kff90\\AppData\\Local\\Programs\\Python\\Python310\\python.exe', 'E:\\stable-diffusion-webui\\extensions\\facechain/facechain/train_text_to_image_lora.py', '--pretrained_model_name_or_path=ly261666/cv_portrait_model', '--revision=v2.0', '--sub_path=film/film', '--output_dataset_name=E:\\stable-diffusion-webui\\extensions\\facechain\\worker_data\\qw\\training_data\\ly261666/cv_portrait_model\\person2', '--caption_column=text', '--resolution=512', '--random_flip', '--train_batch_size=1', '--num_train_epochs=200', '--checkpointing_steps=5000', '--learning_rate=1.5e-04', '--lr_scheduler=cosine', '--lr_warmup_steps=0', '--seed=42', '--output_dir=E:\\stable-diffusion-webui\\extensions\\facechain\\worker_data\\qw\\ly261666/cv_portrait_model\\person2', '--lora_r=4', '--lora_alpha=32', '--lora_text_encoder_r=32', '--lora_text_encoder_alpha=32', '--resume_from_checkpoint=fromfacecommon']' returned non-zero exit status 1.
Error executing the command: Command '['accelerate', 'launch', 'E:\\stable-diffusion-webui\\extensions\\facechain/facechain/train_text_to_image_lora.py', '--pretrained_model_name_or_path=ly261666/cv_portrait_model', '--revision=v2.0', '--sub_path=film/film', '--output_dataset_name=E:\\stable-diffusion-webui\\extensions\\facechain\\worker_data\\qw\\training_data\\ly261666/cv_portrait_model\\person2', '--caption_column=text', '--resolution=512', '--random_flip', '--train_batch_size=1', '--num_train_epochs=200', '--checkpointing_steps=5000', '--learning_rate=1.5e-04', '--lr_scheduler=cosine', '--lr_warmup_steps=0', '--seed=42', '--output_dir=E:\\stable-diffusion-webui\\extensions\\facechain\\worker_data\\qw\\ly261666/cv_portrait_model\\person2', '--lora_r=4', '--lora_alpha=32', '--lora_text_encoder_r=32', '--lora_text_encoder_alpha=32', '--resume_from_checkpoint=fromfacecommon']' returned non-zero exit status 1.
Traceback (most recent call last):
  File "E:\stable-diffusion-webui\extensions\facechain\app.py", line 111, in train_lora_fn
    subprocess.run(command, check=True)
  File "C:\Users\kff90\AppData\Local\Programs\Python\Python310\lib\subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['accelerate', 'launch', 'E:\\stable-diffusion-webui\\extensions\\facechain/facechain/train_text_to_image_lora.py', '--pretrained_model_name_or_path=ly261666/cv_portrait_model', '--revision=v2.0', '--sub_path=film/film', '--output_dataset_name=E:\\stable-diffusion-webui\\extensions\\facechain\\worker_data\\qw\\training_data\\ly261666/cv_portrait_model\\person2', '--caption_column=text', '--resolution=512', '--random_flip', '--train_batch_size=1', '--num_train_epochs=200', '--checkpointing_steps=5000', '--learning_rate=1.5e-04', '--lr_scheduler=cosine', '--lr_warmup_steps=0', '--seed=42', '--output_dir=E:\\stable-diffusion-webui\\extensions\\facechain\\worker_data\\qw\\ly261666/cv_portrait_model\\person2', '--lora_r=4', '--lora_alpha=32', '--lora_text_encoder_r=32', '--lora_text_encoder_alpha=32', '--resume_from_checkpoint=fromfacecommon']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "E:\stable-diffusion-webui\venv\lib\site-packages\gradio\routes.py", line 488, in run_predict
    output = await app.get_blocks().process_api(
  File "E:\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 1431, in process_api
    result = await self.call_function(
  File "E:\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 1103, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "E:\stable-diffusion-webui\venv\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "E:\stable-diffusion-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "E:\stable-diffusion-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "E:\stable-diffusion-webui\venv\lib\site-packages\gradio\utils.py", line 707, in wrapper
    response = f(*args, **kwargs)
  File "E:\stable-diffusion-webui\extensions\facechain\app.py", line 691, in run
    train_lora_fn(base_model_path=base_model_path,
  File "E:\stable-diffusion-webui\extensions\facechain\app.py", line 114, in train_lora_fn
    raise gr.Error("训练失败 (Training failed)")
gradio.exceptions.Error: '训练失败 (Training failed)'

**Comment by wenmengzhou** [comment]
please install mmcv-full==1.7.1

**Comment by KampfschaFF** [comment]
i try the command in powershell:

pip install mmcv-full==1.7.1

AND

E:\stable-diffusion-webui\venv\Scripts\python.exe -m pip install mmcv-full==1.7.1 --prefer-binary

and it stuck in a endless loading wheel

I also tried 

pip uninstall mmcv-full
pip install mmcv-full==1.7.1

with same results

**Comment by Marcophono2** [comment]
@KampfschaFF @wenmengzhou I am wondering why you were able generally to bring Facechain to work under SD 1111. Yesterday I tried everything, including updating every packages to the requested versions but always when I open the SD web ui, I can see the Facechain tab but when I click on it the dashboard below the tabs menu is completely empty.
I found a discussion yesterday about this issues. Are they really solved meanwhile? I mean solved so far that clicking the Facechain tab shows something. :-)

Ubuntu 23.04, 3x RTX 4090, cuda compiler 11.8

Best regards
Marc

**Comment by yamosin** [comment]
I had the same problem and solved it by

Search for whl files on your hard disk and find mmcv_full-1.7.0-cp310-cp310-win_amd64.whl (or 1.7.1) and delete it!
After activating env:
pip uninstall all components: mmdet, mmengine, mmcv, mmcv-full
Go to the facechain folder
pip install -r requirements.txt
pip install modelscope==1.9.2
pip install -U openmim
mim install mmcv-full==1.7.0
This will take a long time to compile.

Make sure there are no incorrect dependencies and recompile mmcv-full for the corresponding torch/cuda version (I wasted an hour on this because mmcv-full uses the cached mmcv_full-1.7.0-cp310-cp310-win_amd64.whl and I don't know why it's having problems. (because it also compiles this way)
Also I noticed that your error message has "module named 'datasets'" "No module named 'triton'".
Try installing them with pip install
